rita_client/src/exit_manager/mod.rs

```diff
@@ -562,6 +555,26 @@ fn correct_default_route(input: Option<DefaultRoute>) -> bool {
     }
 }
 
+/// This function takes an exit server and chooses the exit ip to return
+/// When the exit server provided is the one we are currently connected to, there is no guarantee
+/// that the exit ip in use is the one we are connected to, so we return get_current_selected_exit()
+/// When the exit server is not the one we are connected to, we return subnet.ip(), which is the default
+/// ip address of that exit. This is used in cases where we need to get general information about an exit for
```

Comment:
I think this needs to be split into two functions, one to get the current exit ip and one to get the best ip for a given exit passed to the function.

as it is this function behaves differently depending on the situation and may cause strange behavior if for example we don't have an exit selected.